cache/src/main/java/com/uber/buckcache/datastore/impl/ignite/IgniteConfigurationBuilder.java

```diff
@@ -28,12 +32,25 @@ public IgniteConfigurationBuilder() {
     igniteConfiguration = new IgniteConfiguration();
   }
 
-  public IgniteConfigurationBuilder addMulticastBasedDiscrovery(String multicastIP, Integer multicastPort, List<String> hostIPs) {
+  public IgniteConfigurationBuilder addMulticastBasedDiscrovery(String multicastIP, Integer multicastPort, List<String> hostIPs, String dnsLookupAddress) {
     TcpDiscoverySpi spi = new TcpDiscoverySpi();
+
     TcpDiscoveryVmIpFinder ipFinder = new TcpDiscoveryMulticastIpFinder();
     ((TcpDiscoveryMulticastIpFinder) ipFinder).setMulticastGroup(multicastIP);
     ((TcpDiscoveryMulticastIpFinder) ipFinder).setMulticastPort(multicastPort);
-    ((TcpDiscoveryMulticastIpFinder) ipFinder).setAddresses(hostIPs);
+
+    // not allow providing both hostIPs and dnsLookupAddress
+    boolean hasHostIPs = hostIPs != null && !hostIPs.isEmpty();
+    boolean hasDNSLookupAddress = !Strings.isNullOrEmpty(dnsLookupAddress);
+    if (hasHostIPs && hasDNSLookupAddress) {
```

Comment:
I dont think this should be a requirement. Any reason why we cannot add all the hosts specified via various means to the pool?